modules/statistics/RTPStatsCollector.js

```diff
@@ -552,29 +577,34 @@ StatsCollector.prototype.processStatsReport = function() {
                 });
             }
 
-            // Get the resolution and framerate for only remote video sources here. For the local video sources,
-            // 'track' stats will be used since they have the updated resolution based on the simulcast streams
-            // currently being sent. Promise based getStats reports three 'outbound-rtp' streams and there will be
-            // more calculations needed to determine what is the highest resolution stream sent by the client if the
-            // 'outbound-rtp' stats are used.
-            if (now.type === 'inbound-rtp') {
-                const resolution = {
-                    height: now.frameHeight,
-                    width: now.frameWidth
-                };
-                const frameRate = now.framesPerSecond;
-
-                if (resolution.height && resolution.width) {
-                    ssrcStats.setResolution(resolution);
-                }
+            const resolution = {
+                height: now.frameHeight,
+                width: now.frameWidth
+            };
+            const frameRate = now.framesPerSecond;
+
+            // Process the stats for 'inbound-rtp' streams always and 'outbound-rtp' only if the browser is
+            // Chromium based and version 112 and later since 'track' based stats are no longer available there
+            // for calculating send resolution and frame rate.
+            if (resolution.height
+                && resolution.width
+                && (now.type === 'inbound-rtp'
+                || (!browser.supportsTrackBasedStats() && now.active))) {
+                ssrcStats.setResolution(resolution);
                 ssrcStats.setFramerate(Math.round(frameRate || 0));
 
-                if (before) {
-                    ssrcStats.addBitrate({
-                        'download': this._calculateBitrate(now, before, 'bytesReceived'),
-                        'upload': 0
-                    });
-                }
+            // Reset the stats if the current stream is suspended when 'outbound-rtp' stats are used for resolution
+            // and framerate.
+            } else if (!browser.supportsTrackBasedStats()) {
```

Comment:
If i'm reading this right you could end up here if `resolution.height` is 0 above, even if the type is `inbound-rtp`.

Reply:
Yup, l will fix that.

Reply:
IMHO it makes the code less readable because you need to know what you just said for it to be correct. I'd rather the code be explicit.

Reply:
Ok, I addressed this in the most recent commit.